### Data
A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29
### Assignment Tasks
a. Build 99% Confidence Interval Using Sample Standard Deviation
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

b. Build 99% Confidence Interval Using Known Population Standard Deviation
If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.


In [1]:
import pandas as pd 
import numpy as np
#from scipy import stats
import matplotlib.pyplot as plt
#from statsmodels.stats.weightstats import ztest
from scipy.stats import t, norm


In [2]:
data=np.array([ 1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
data

array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.2 ,
       1.33, 1.18, 1.22, 1.29])

## a. Build 99% Confidence Interval Using Sample Standard Deviation

In [5]:
#Sample statistics
sample_mean = np.mean(data)
sample_std = np.std(data, ddof=1)  # ddof=1 for sample standard deviation
n = len(data)
df = n - 1

In [12]:
sample_mean
print(sample_std)

0.19316412956959936


In [6]:
# a. Confidence Interval using Sample Standard Deviation
confidence_level = 0.99
alpha = 1 - confidence_level



In [7]:
# Find t-score
t_score = t.ppf(1 - alpha / 2, df)


In [8]:
# Calculate Margin of Error
margin_of_error = t_score * (sample_std / np.sqrt(n))


In [9]:
# Calculate Confidence Interval
lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error


In [10]:
print("a. Confidence Interval using Sample Standard Deviation:")
print("Sample Mean:", sample_mean)
print("Sample Standard Deviation:", sample_std)
print("Degrees of Freedom:", df)
print("T-Score:", t_score)
print("Margin of Error:", margin_of_error)
print("Confidence Interval:", (lower_bound, upper_bound))


a. Confidence Interval using Sample Standard Deviation:
Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936
Degrees of Freedom: 14
T-Score: 2.97684273411266
Margin of Error: 0.1484693282152996
Confidence Interval: (1.090197338451367, 1.3871359948819662)


## hence the  99% Confidence Interval Using Sample Standard Deviation is  (1.090197338451367, 1.3871359948819662)

## b. Build 99% Confidence Interval Using Known Population Standard Deviation

In [15]:
population_std = 0.2  # given population standard deviation


In [16]:
# Calculate z-score (since we know population standard deviation)
z_score = norm.ppf(1 - alpha / 2)



In [17]:
# Calculate Margin of Error
margin_of_error_known_std = z_score * (population_std / np.sqrt(n))


In [18]:
# Calculate Confidence Interval
lower_bound_known_std = sample_mean - margin_of_error_known_std
upper_bound_known_std = sample_mean + margin_of_error_known_std


In [19]:
print("\nb. Confidence Interval using Known Population Standard Deviation:")
print("Sample Mean:", sample_mean)
print("Population Standard Deviation:", population_std)
print("Z-Score:", z_score)
print("Margin of Error:", margin_of_error_known_std)
print("Confidence Interval:", (lower_bound_known_std, upper_bound_known_std))


b. Confidence Interval using Known Population Standard Deviation:
Sample Mean: 1.2386666666666666
Population Standard Deviation: 0.2
Z-Score: 2.5758293035489004
Margin of Error: 0.13301525327090588
Confidence Interval: (1.1056514133957607, 1.3716819199375725)


## hence  99% Confidence Interval Using 0.2 Population Standard Deviation is(1.1056514133957607, 1.3716819199375725)
